### PR TITLE
feat: training segmentation models with torch backend

### PR DESCRIPTION
**Pull request by beniz**
This PR brings segmentation model training to the torch backend.

### CODE REVIEW

File: src/backends/torch/torchdataset.cc [hunk 1]
```diff
@@ -173,8 +200,20 @@ namespace dd
     bgr = cv::Mat(img_data, true);
     bgr = cv::imdecode(bgr,
                        bw ? CV_LOAD_IMAGE_GRAYSCALE : CV_LOAD_IMAGE_COLOR);
-    std::stringstream targetstream(targets);
-    torch::load(targett, targetstream);
+    cv::Mat bw_target; // for segmentation only.
+
+    if (!_segmentation)
```
Comment (Bycob) [inline src/backends/torch/torchdataset.cc:205]: Invert?
```
if (_segmentation) {
...
} else {
...
}
```
It feels more natural

File: src/backends/torch/torchdataset.h [hunk 2]
```diff
@@ -309,7 +324,7 @@ namespace dd
      * \brief turns an image into a torch::Tensor
      */
     at::Tensor image_to_tensor(const cv::Mat &bgr, const int &height,
-                               const int &width);
+                               const int &width, const bool &target = false);
```
Comment (Bycob) [inline src/backends/torch/torchdataset.h:327]: Add documentation for param `target`?

File: src/backends/torch/torchlib.cc [hunk 3]
```diff
@@ -905,8 +905,16 @@ namespace dd
                 Tensor y = batch.target.at(0).to(device);
 
                 // Prediction
-                Tensor y_pred = torch_utils::to_tensor_safe(
-                    rank_module.forward(in_vals));
+                Tensor y_pred;
+                if (!_segmentation)
```
Comment (Bycob) [inline src/backends/torch/torchlib.cc:909]:
```
if (seg)
else
```

File: src/backends/torch/torchdataset.cc (hunk 1 above)
Comment (beniz) [reply src/backends/torch/torchdataset.cc:205]: OK

File: src/backends/torch/torchdataset.h (hunk 2 above)
Comment (beniz) [reply src/backends/torch/torchdataset.h:331]: ok, done.

File: src/backends/torch/torchlib.cc (hunk 3 above)
Comment (beniz) [reply src/backends/torch/torchlib.cc]: OK done

**Review by Bycob — APPROVED** [review]

**Review by fantes — APPROVED** [review]

**Pull request merged**